src/MetaMorpho.sol

```diff
@@ -628,7 +647,7 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
 
         _supplyMorpho(assets);
 
-        // `newTotalAssets + assets` cannot be used as input because of rounding errors so we must use `totalAssets`.
+        // Update `lastTotalAssets` again to avoid inconsistent state due to a reentrancy for instance.
         _updateLastTotalAssets(totalAssets());
```

Comment:
```suggestion
        // `lastTotalAssets + assets` may be a little off from `totalAssets()`.
        _updateLastTotalAssets(lastTotalAssets + assets);
```

Reply:
It's not symmetric deposit/mint then and introduce small rounding. Are we sure we want to do that?

Reply:
> It's not symmetric deposit/mint

I don't get the symmetry you're talking about ; to me there's no symmetry nowhere

> introduce small rounding

Yes, as in withdraw/redeem
If we don't want this rounding in the end, we should revert the change in withdraw/redeem (and keep the previous comments about why we're doing so)

Reply:
In `mint`/`deposit`/`withdraw`/`redeem` the `lastTotalAssets` is updated using `totalAssets()` ultimately which would not be the case anymore with your proposition for `mint`/`deposit` only

Reply:
That is incorrect: with the current changes to withdraw and redeem, lastTotalAssets is no longer calculated using totalAssets()

Hence why I'm suggesting to add assets instead of calling totalAssets() in deposit/mint: to be consistent
But we can be consistent the other way around by reverting the changes to withdraw & redeem

Reply:
This is the current state of the code:

https://github.com/morpho-org/metamorpho/blob/4220004a6a38caacaff31a8af0e67a1ccdef5842/src/MetaMorpho.sol#L651

Reply:
Yes and I'm highlighting that there's no symmetry with withdraw/redeem: https://github.com/morpho-org/metamorpho/blob/4220004a6a38caacaff31a8af0e67a1ccdef5842/src/MetaMorpho.sol#L568-L569